Convert UTC to BST

sc15zs opened the pull request:
Description:
A filter to convert datetimes stored in UTC to BST (British Standard Time)

albertkol reviewed (commented):
Please use the existing `utc_to_bst` filter from `pre_award/assess/shared/filters.py`

albertkol reviewed (changes requested):
You are missing the conversion in more places:
- `pre_award/authenticator/frontend/templates/authenticator/magic_links/landing.html`, line 21 should be: 
```
{% trans %}Submission deadline:{% endtrans %} <span class="govuk-body">{{ submission_deadline|string_to_datetime|to_bst|datetime_format_respect_lang }}</span>
```
- `pre_award/apply/templates/apply/tasklist.html` line 114 should be:
```
<dd class="govuk-body govuk-!-margin-bottom-2"> {{ submission_deadline|string_to_datetime|to_bst|datetime_format_full_month }}</dd>
```

Could you please also add tests to the new bst conversion function? Similar to what @sfount did at `tests/pre_award/assess_tests/test_filters.py`, `test_utc_to_bst`.

albertkol commented:
> Please use the existing `utc_to_bst` filter from `pre_award/assess/shared/filters.py`

We won't use the `utc_to_bst` as it's not doing exactly what we need for Apply. Also we seem to favour re-creating fresh functions in the `common` dir, similar to what we did for `datetime_format_respect_lang`.

sc15zs commented on common/utils/filters.py:
```diff
@@ -38,3 +39,13 @@ def datetime_format_respect_lang(value: datetime) -> str:
+
+def to_bst(value: datetime) -> datetime:
+    """Converts a datetime object to British Summer Time (BST)."""
+    if value:
```
Changed the function as requested.

sc15zs commented on common/utils/filters.py:
```diff
@@ -38,3 +39,13 @@ def datetime_format_respect_lang(value: datetime) -> str:
+    """Converts a datetime object to British Summer Time (BST)."""
+    if value:
+        utc = pytz.utc
+        bst = pytz.timezone("Europe/London")
```
Done.

sc15zs commented:
> You are missing the conversion in more places:
> 
> * `pre_award/authenticator/frontend/templates/authenticator/magic_links/landing.html`, line 21 should be:
> 
> ```
> {% trans %}Submission deadline:{% endtrans %} <span class="govuk-body">{{ submission_deadline|string_to_datetime|to_bst|datetime_format_respect_lang }}</span>
> ```
> 
> * `pre_award/apply/templates/apply/tasklist.html` line 114 should be:
> 
> ```
> <dd class="govuk-body govuk-!-margin-bottom-2"> {{ submission_deadline|string_to_datetime|to_bst|datetime_format_full_month }}</dd>
> ```
> 
> Could you please also add tests to the new bst conversion function? Similar to what @sfount did at `tests/pre_award/assess_tests/test_filters.py`, `test_utc_to_bst`.

Added the new filter to the templates, and added a unit test.

ksp37-dluhc commented on tests/pre_award/apply_tests/test_filters.py:
```diff
@@ -33,6 +36,32 @@ def test_datetime_format(input_date, expected, app):
+        (
+            dt(2025, 9, 30, 1, 25, 0),
+            timezone("Europe/London").localize(dt(2025, 9, 30, 2, 25, 0)),
+        ),  # When BST is one hour ahead of UTC
```
Is it worth adding one test case where `input_date` is a datetime with an existing timezone? 

I'm not sure if it will ever be used this way, but it looks like it's handled in the function.

sc15zs replied:
Thanks, Kishan. Added 2 test cases with UTC input date and one with BST.
